diff --git a/lib/Basic/Version.cpp b/lib/Basic/Version.cpp
index f15f4d63be0b..2bd76a0e89d9 100644
--- a/lib/Basic/Version.cpp
+++ b/lib/Basic/Version.cpp
@@ -409,31 +409,7 @@ std::string getSwiftFullVersion(Version effectiveVersion) {
   OS << SWIFT_VENDOR " ";
 #endif
 
-  OS << "Swift version " SWIFT_VERSION_STRING;
-#ifndef SWIFT_COMPILER_VERSION
-  OS << "-dev";
-#endif
-
-  if (!(effectiveVersion == Version::getCurrentLanguageVersion())) {
-    OS << " effective-" << effectiveVersion;
-  }
-
-#if defined(SWIFT_COMPILER_VERSION)
-  OS << " (swiftlang-" SWIFT_COMPILER_VERSION;
-#if defined(CLANG_COMPILER_VERSION)
-  OS << " clang-" CLANG_COMPILER_VERSION;
-#endif
-  OS << ")";
-#elif defined(LLVM_REVISION) || defined(CLANG_REVISION) || \
-      defined(SWIFT_REVISION)
-  OS << " (";
-  printFullRevisionString(OS);
-  OS << ")";
-#endif
-
-  // Suppress unused function warning
-  (void)&printFullRevisionString;
-
+  OS << "Swift version " SWIFT_VERSION_STRING " (swift-arm-oss-RELEASE)";
   return OS.str();
 }
 
