diff --git a/lib/Driver/UnixToolChains.cpp b/lib/Driver/UnixToolChains.cpp
index 61498c144a2..af8e9ed89ca 100644
--- a/lib/Driver/UnixToolChains.cpp
+++ b/lib/Driver/UnixToolChains.cpp
@@ -113,6 +113,10 @@ ToolChain::InvocationInfo toolchains::GenericUnix::constructInvocation(
 std::string toolchains::GenericUnix::getDefaultLinker() const {
   if (getTriple().isAndroid())
     return "lld";
+  // Check for Amazon Linux and default to lld
+  if (getTriple().getVendorName() == "amazon")
+    std::cout << "We are using Amazon Linux so default to lld\n";
+    return "lld";
 
   switch (getTriple().getArch()) {
   case llvm::Triple::arm:
