diff --git a/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift b/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift
index 1fcf6f36..1b25ef15 100644
--- a/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift
+++ b/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift
@@ -28,10 +28,10 @@ extension GenericUnixToolchain {
       // section on these targets, it also generates COPY relocations for
       // final executables, as such, unless specified, we default to gold
       // linker.
-      return "gold"
+      return "lld"
     case .x86, .x86_64, .ppc64, .ppc64le, .systemz:
       // BFD linker has issues wrt relocations against protected symbols.
-      return "gold"
+      return "lld"
     default:
       // Otherwise, use the default BFD linker.
       return ""
