diff --git a/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift b/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift
index 518c6e95..4812b8ec 100644
--- a/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift
+++ b/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift
@@ -21,7 +21,20 @@ extension GenericUnixToolchain {
         targetTriple.environment == .android {
       return "lld"
     }
-
+  func readOSReleaseID() -> String? {
+    let osReleasePath = AbsolutePath("/etc/os-release")
+    guard let fileContents = try? localFileSystem.readFileContents(osReleasePath).cString,
+          let fileContentsString = String(cString: fileContents).split(separator: "\n").first(where: { $0.hasPrefix("ID=") }),
+          let id = fileContentsString.dropFirst("ID=".count) else {
+      return nil
+    }
+    return String(id)
+  }
+  if readOSReleaseID() == "amzn" {
+    print("Amazon Linux detected, using lld linker")
+    // Amazon Linux 2023 no longer has "gold" linker and requires "lld" linker.
+    return "lld"
+  }
     switch targetTriple.arch {
     case .arm, .aarch64, .armeb, .thumb, .thumbeb:
       // BFD linker has issues wrt relocation of the protocol conformance
