diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index 512664e..cbe9e3b 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -782,6 +782,49 @@ skip-test-libdispatch
 skip-test-playgroundsupport
 skip-test-libicu
 
+# Linux Individual Tests
+[preset: buildbot_linux,test_libdispatch]
+mixin-preset=buildbot_linux
+
+skip-test-cmark
+skip-test-lldb
+skip-test-swift
+skip-test-llbuild
+skip-test-swiftpm
+skip-test-xctest
+skip-test-foundation
+#skip-test-libdispatch
+skip-test-playgroundsupport
+skip-test-libicu
+
+[preset: buildbot_linux,test_foundation]
+mixin-preset=buildbot_linux
+
+skip-test-cmark
+skip-test-lldb
+skip-test-swift
+skip-test-llbuild
+skip-test-swiftpm
+skip-test-xctest
+#skip-test-foundation
+skip-test-libdispatch
+skip-test-playgroundsupport
+skip-test-libicu
+
+[preset: buildbot_linux,test_swiftpm]
+mixin-preset=buildbot_linux
+
+skip-test-cmark
+skip-test-lldb
+skip-test-swift
+skip-test-llbuild
+#skip-test-swiftpm
+skip-test-xctest
+skip-test-foundation
+skip-test-libdispatch
+skip-test-playgroundsupport
+skip-test-libicu
+
 # Linux package with out test
 [preset: buildbot_linux,no_test]
 mixin-preset=
