diff --git a/test/ClangImporter/cfuncs_parse.swift b/test/ClangImporter/cfuncs_parse.swift
index e5e2cb1..70646c8 100644
--- a/test/ClangImporter/cfuncs_parse.swift
+++ b/test/ClangImporter/cfuncs_parse.swift
@@ -1,4 +1,5 @@
 // RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -verify -I %S/Inputs %s
+// REQUIRES: CPU=x86_64
 
 import cfuncs
 
diff --git a/test/Constraints/diagnostics.swift b/test/Constraints/diagnostics.swift
index 4a10050..deb22fe 100644
--- a/test/Constraints/diagnostics.swift
+++ b/test/Constraints/diagnostics.swift
@@ -1,4 +1,5 @@
 // RUN: %target-typecheck-verify-swift
+// REQUIRES: CPU=x86_64
 
 protocol P {
   associatedtype SomeType
diff --git a/test/Driver/fuzzer.swift b/test/Driver/fuzzer.swift
index 3903892..31ea13c 100644
--- a/test/Driver/fuzzer.swift
+++ b/test/Driver/fuzzer.swift
@@ -1,4 +1,5 @@
 // UNSUPPORTED: win32
+// REQUIRES: CPU=x86_64
 // RUN: %swiftc_driver -driver-print-jobs -sanitize=fuzzer,address -resource-dir %S/Inputs/fake-resource-dir/lib/swift/ %s | %FileCheck -check-prefix=LIBFUZZER %s
 
 // LIBFUZZER: libclang_rt.fuzzer
diff --git a/test/IRGen/pic.swift b/test/IRGen/pic.swift
index dc9ba80..c6efa42 100644
--- a/test/IRGen/pic.swift
+++ b/test/IRGen/pic.swift
@@ -2,6 +2,7 @@
 // platforms.
 
 // RUN: %target-swift-frontend -assume-parsing-unqualified-ownership-sil %s -module-name main -S -o - | %FileCheck -check-prefix=%target-cpu %s
+// REQUIRES: CPU=x86_64
 
 var global: Int = 0
 
diff --git a/test/Runtime/linux-fatal-backtrace.swift b/test/Runtime/linux-fatal-backtrace.swift
index 6df5ec3..0d3db63 100644
--- a/test/Runtime/linux-fatal-backtrace.swift
+++ b/test/Runtime/linux-fatal-backtrace.swift
@@ -4,6 +4,7 @@
 // REQUIRES: executable_test
 // REQUIRES: OS=linux-gnu
 // REQUIRES: lldb
+// REQUIRES: CPU=x86_64
 
 // Backtraces are not emitted when optimizations are enabled. This test can not
 // run when optimizations are enabled.
