diff --git a/Sources/SKCore/Toolchain.swift b/Sources/SKCore/Toolchain.swift
index 822a736..5197fc6 100644
--- a/Sources/SKCore/Toolchain.swift
+++ b/Sources/SKCore/Toolchain.swift
@@ -116,7 +116,8 @@ extension Toolchain {
     return
       searchForTools(binPath: path, fs) ||
       searchForTools(binPath: path.appending(components: "bin"), fs) ||
-      searchForTools(binPath: path.appending(components: "usr", "bin"), fs)
+      searchForTools(binPath: path.appending(components: "usr", "bin"), fs) ||
+      searchForTools(binPath: path.appending(components: "usr", "lib", "swift", "bin"), fs)
   }
 
   private func searchForTools(binPath: AbsolutePath, _ fs: FileSystem) -> Bool {
