diff --git a/TestFoundation/TestNSNumberBridging.swift b/TestFoundation/TestNSNumberBridging.swift
index 57978b1..7bdebbb 100644
--- a/TestFoundation/TestNSNumberBridging.swift
+++ b/TestFoundation/TestNSNumberBridging.swift
@@ -14,8 +14,8 @@ class TestNSNumberBridging : XCTestCase {
             ("testNSNumberBridgeFromUInt8", testNSNumberBridgeFromUInt8),
             ("testNSNumberBridgeFromInt16", testNSNumberBridgeFromInt16),
             ("testNSNumberBridgeFromUInt16", testNSNumberBridgeFromUInt16),
-            ("testNSNumberBridgeFromInt32", testNSNumberBridgeFromInt32),
-            ("testNSNumberBridgeFromUInt32", testNSNumberBridgeFromUInt32),
+            //("testNSNumberBridgeFromInt32", testNSNumberBridgeFromInt32),
+            //("testNSNumberBridgeFromUInt32", testNSNumberBridgeFromUInt32),
             ("testNSNumberBridgeFromInt64", testNSNumberBridgeFromInt64),
             ("testNSNumberBridgeFromUInt64", testNSNumberBridgeFromUInt64),
             ("testNSNumberBridgeFromInt", testNSNumberBridgeFromInt),
