diff --git a/Sources/Build/Triple.swift b/Sources/Build/Triple.swift
index 1ce1556..4a9d2e5 100644
--- a/Sources/Build/Triple.swift
+++ b/Sources/Build/Triple.swift
@@ -32,8 +32,11 @@ public struct Triple {
 
     public enum Arch: String {
         case x86_64
-        case armv7
+        case ppc64le
         case s390x
+        case aarch64
+        case armv7
+        case arm
     }
 
     public enum Vendor: String {
@@ -104,14 +107,26 @@ public struct Triple {
     }
 
     public static let macOS = try! Triple("x86_64-apple-macosx10.10")
-    public static let linux = try! Triple("x86_64-unknown-linux")
+    public static let x86Linux = try! Triple("x86_64-unknown-linux")
+    public static let ppc64leLinux = try! Triple("powerpc64le-unknown-linux")
+    public static let s390xLinux = try! Triple("s390x-unknown-linux")
+    public static let arm64Linux = try! Triple("aarch64-unknown-linux")
+    public static let armLinux = try! Triple("armv7-unknown-linux-gnueabihf")
     public static let android = try! Triple("armv7-unknown-linux-androideabi")
 
   #if os(macOS)
     public static let hostTriple: Triple = .macOS
-  #elseif os(Linux) && arch(s390x)
-    public static let hostTriple: Triple = try! Triple("s390x-unknown-linux")
-  #else
-    public static let hostTriple: Triple = .linux
+  #elseif os(Linux)
+    #if arch(x86_64)
+      public static let hostTriple: Triple = .x86Linux
+    #elseif arch(powerpc64le)
+      public static let hostTriple: Triple = .ppc64leLinux
+    #elseif arch(s390x)
+      public static let hostTriple: Triple = .s390xLinux
+    #elseif arch(arm64)
+      public static let hostTriple: Triple = .arm64Linux
+    #elseif arch(arm)
+      public static let hostTriple: Triple = .armLinux
+    #endif
   #endif
 }
diff --git a/Utilities/bootstrap b/Utilities/bootstrap
index 784a232..6ae5e98 100755
--- a/Utilities/bootstrap
+++ b/Utilities/bootstrap
@@ -891,10 +891,21 @@ def main():
     # Compute the build paths.
     if platform.system() == 'Darwin':
         build_target = "x86_64-apple-macosx10.10"
-    elif platform.processor() == 's390x':
-        build_target = "s390x-unknown-linux"
+    elif platform.system() == 'Linux':
+        if platform.machine() == 'x86_64':
+            build_target = "x86_64-unknown-linux"
+        elif platform.machine() == 's390x':
+            build_target = "s390x-unknown-linux"
+        elif platform.machine() == 'ppc64le':
+            build_target = 'powerpc64le-unknown-linux'
+        elif platform.machine().startswith("armv7"):
+            build_target = 'armv7-unknown-linux-gnueabihf'
+        elif platform.machine() == 'aarch64':
+            build_target = 'aarch64-unknown-linux'
+        else:
+            raise SystemExit("ERROR: unsupported machine:",platform.machine())
     else:
-        build_target = 'x86_64-unknown-linux'
+        raise SystemExit("ERROR: unsupported system:",platform.system())
 
     build_path = os.path.join(g_project_root, args.build_path)
     sandbox_path = os.path.join(build_path, ".bootstrap")
