diff --git a/Sources/Build/Triple.swift b/Sources/Build/Triple.swift
index fcbeb2df..e41efaec 100644
--- a/Sources/Build/Triple.swift
+++ b/Sources/Build/Triple.swift
@@ -136,7 +136,7 @@ public struct Triple: Encodable {
     public static let i686Linux = try! Triple("i686-unknown-linux")
     public static let ppc64leLinux = try! Triple("powerpc64le-unknown-linux")
     public static let s390xLinux = try! Triple("s390x-unknown-linux")
-    public static let arm64Linux = try! Triple("aarch64-unknown-linux")
+    public static let arm64Linux = try! Triple("aarch64-unknown-linux-gnu")
     public static let armLinux = try! Triple("armv7-unknown-linux-gnueabihf")
     public static let armAndroid = try! Triple("armv7a-unknown-linux-androideabi")
     public static let arm64Android = try! Triple("aarch64-unknown-linux-android")
