diff --git a/third_party/gpus/crosstool/BUILD.tpl b/third_party/gpus/crosstool/BUILD.tpl
index 0c5abf9..856a265 100644
--- a/third_party/gpus/crosstool/BUILD.tpl
+++ b/third_party/gpus/crosstool/BUILD.tpl
@@ -28,6 +28,7 @@ cc_toolchain_suite(
         "piii": ":cc-compiler-local",
         "ppc": ":cc-compiler-local",
         "darwin": ":cc-compiler-darwin",
+        "aarch64": ":cc-compiler-local",
     },
 )

