diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index 477fc17..3f20d47 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -1877,21 +1877,25 @@ darwin-toolchain-installer-package=%(darwin_toolchain_installer_package)s
 mixin-preset=buildbot_linux
 enable-tensorflow
 install-tensorflow
+tensorflow_bazel_options=--define=tensorflow_mkldnn_contraction_kernel=0
 
 [preset: tensorflow_linux,no_test]
 mixin-preset=buildbot_linux,no_test
 enable-tensorflow
 install-tensorflow
+tensorflow_bazel_options=--define=tensorflow_mkldnn_contraction_kernel=0
 
 [preset: tensorflow_linux,gpu]
 mixin-preset=tensorflow_linux
 enable-tensorflow-gpu
 install-tensorflow
+tensorflow_bazel_options=--define=tensorflow_mkldnn_contraction_kernel=0
 
 [preset: tensorflow_linux,gpu,no_test]
 mixin-preset=tensorflow_linux,no_test
 enable-tensorflow-gpu
 install-tensorflow
+tensorflow_bazel_options=--define=tensorflow_mkldnn_contraction_kernel=0
 
 #===------------------------------------------------------------------------===#
 # Swift for TensorFlow Preset
