diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index 18530bf..8ac1aa2 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -805,6 +805,12 @@ android-icu-uc-include=%(arm_dir)s/icu/source/common
 android-icu-i18n=%(arm_dir)s/libicui18nswift.so
 android-icu-i18n-include=%(arm_dir)s/icu/source/i18n
 
+# Ubuntu 16.04 preset for Tensorflow on AArch64.
+[preset: buildbot_linux_1604_tensorflow]
+mixin-preset=buildbot_linux,no_test
+enable-tensorflow
+tensorflow_bazel_options=--define=tensorflow_mkldnn_contraction_kernel=0
+
 # Ubuntu 18.04 preset for backwards compat and future customizations.
 [preset: buildbot_linux_1804]
 mixin-preset=buildbot_linux
