diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index 496b97e..5884b9b 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -746,7 +746,7 @@ build-swift-stdlib-unittest-extra
 # Executes the lit tests for the installable package that is created
 # Assumes the swift-integration-tests repo is checked out
 
-test-installable-package
+# test-installable-package
 
 # Build the benchmarks against the toolchain.
 toolchain-benchmarks
@@ -2126,25 +2126,25 @@ darwin-toolchain-installer-package=%(darwin_toolchain_installer_package)s
 mixin-preset=buildbot_linux
 enable-tensorflow
 install-tensorflow
-# The swift-package-tests fail when run with python3.
-test-installable-package=
+tensorflow_bazel_options=--define=tensorflow_mkldnn_contraction_kernel=0
 
 [preset: tensorflow_linux,no_test]
 mixin-preset=buildbot_linux,no_test
 enable-tensorflow
 install-tensorflow
-# The swift-package-tests fail when run with python3.
-test-installable-package=
+tensorflow_bazel_options=--define=tensorflow_mkldnn_contraction_kernel=0
 
 [preset: tensorflow_linux,gpu]
 mixin-preset=tensorflow_linux
 enable-tensorflow-gpu
 install-tensorflow
+tensorflow_bazel_options=--define=tensorflow_mkldnn_contraction_kernel=0
 
 [preset: tensorflow_linux,gpu,no_test]
 mixin-preset=tensorflow_linux,no_test
 enable-tensorflow-gpu
 install-tensorflow
+tensorflow_bazel_options=--define=tensorflow_mkldnn_contraction_kernel=0
 
 #===------------------------------------------------------------------------===#
 # Swift for TensorFlow Preset
