diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index f640950..5205c1d 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -1502,6 +1502,7 @@ toolchains::GenericUnix::constructInvocation(const AutolinkExtractJobAction &job
 std::string toolchains::GenericUnix::getDefaultLinker() const {
   switch(getTriple().getArch()) {
   case llvm::Triple::arm:
+  case llvm::Triple::aarch64:
   case llvm::Triple::armeb:
   case llvm::Triple::thumb:
   case llvm::Triple::thumbeb:
